azurerm/internal/services/network/express_route_connection_resource.go

```diff
@@ -0,0 +1,349 @@
+							Type:       schema.TypeList,
+							Optional:   true,
+							Computed:   true,
+							MaxItems:   1,
+							ConfigMode: schema.SchemaConfigModeAttr,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"labels": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validation.StringIsNotEmpty,
+										},
+									},
+
+									"route_table_ids": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validate.HubRouteTableID,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"routing_weight": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      0,
+				ValidateFunc: validation.IntBetween(0, 32000),
+			},
+		},
+	}
+}
+
+func resourceExpressRouteConnectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteConnectionsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	expressRouteGatewayId, err := parse.ExpressRouteGatewayID(d.Get("express_route_gateway_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id := parse.NewExpressRouteConnectionID(expressRouteGatewayId.SubscriptionId, expressRouteGatewayId.ResourceGroup, expressRouteGatewayId.Name, d.Get("name").(string))
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.ExpressRouteGatewayName, id.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_express_route_connection", id.ID())
+		}
+	}
+
+	expressRouteConnectionParameters := network.ExpressRouteConnection{
+		Name: utils.String(id.Name),
+		ExpressRouteConnectionProperties: &network.ExpressRouteConnectionProperties{
+			ExpressRouteCircuitPeering: &network.ExpressRouteCircuitPeeringID{
+				ID: utils.String(d.Get("express_route_circuit_peering_id").(string)),
+			},
+			RoutingWeight: utils.Int32(int32(d.Get("routing_weight").(int))),
+		},
+	}
+
+	if v, ok := d.GetOk("authorization_key"); ok {
+		expressRouteConnectionParameters.ExpressRouteConnectionProperties.AuthorizationKey = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("enable_internet_security"); ok {
+		expressRouteConnectionParameters.ExpressRouteConnectionProperties.EnableInternetSecurity = utils.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("routing"); ok {
+		expressRouteConnectionParameters.ExpressRouteConnectionProperties.RoutingConfiguration = expandExpressRouteConnectionRouting(v.([]interface{}))
```

Comment:
Can we inline this when initializing the `expressRouteConnectionParameters`?

Reply:
Updated